src/main.rs

```diff
@@ -39,17 +41,22 @@ fn run_script(path: &str) {
     println!("output\n{}", str::from_utf8(&output.stdout).unwrap());
 }
 
+fn str_to_u8(s: &str) -> Result<Vec<u8>, ParseIntError> {
+    (5..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
```

Comment:
If the request is malformed then this index operation could panic.

Note that you don't have to be constant time relative to public information such as whether the encoding is a valid hex-formatted version of the hmac, you only have to be constant time with regards to the comparison the your secret.

Reply:
@HeroicKatora 
I just made a change so that it checks the length first. 
```
if s.len() != 45 {
        return Err("Invalid x-hub-signature string")
    }
```
So this should prevent the index operation error. Is there anything that I overlooked?

Reply:
I don't think so, the fix seems good.

Reply:
Thank you very much for your time!